nengo/builder/signal.py

```diff
@@ -51,12 +51,19 @@ def __init__(self, initial_value,
             assert name
         self._name = name
 
-        initial_value = np.asarray(initial_value)
-        if initial_value.ndim > 0 and base is None:
-            self._initial_value = np.ascontiguousarray(initial_value).view()
+        self._initial_value = initial_value
+        if self.type == "dense":
```

Comment:
How can `self.type` be already set here?

Reply:
`self.type` is a property